src/taskgraph/generator.py

```diff
@@ -31,22 +31,47 @@ class KindNotFound(Exception):
     """
 
 
+# In an ideal world these would be part of each loader. Unfortunately, loaders
+# can only provide information as part of yielding task inputs, and we need to
+# know the set of transforms prior to starting to read from the loader.
+DEFAULT_TRANSFORMS = [
+    "taskgraph.transforms.job:transforms",
+    "taskgraph.transforms.task:transforms",
+]
+
+
 @attr.s(frozen=True)
 class Kind:
     name = attr.ib(type=AnyStr)
     path = attr.ib(type=AnyStr)
     config = attr.ib(type=dict)
     graph_config = attr.ib(type=GraphConfig)
 
-    def _get_loader(self):
+    def _get_loader_and_transforms(self):
+        transform_refs = copy.copy(self.config.get("transforms", []))
         try:
             loader = self.config["loader"]
         except KeyError:
-            raise KeyError(f"{self.path!r} does not define `loader`")
-        return find_object(loader)
+            # To avoid confusion, we do not allow these transforms to be
+            # specified if they are going to be added automatically. This
+            # avoids issues such as including them twice, being in a different
+            # order in the kind vs. here, etc.
+            for t in DEFAULT_TRANSFORMS:
+                if t in transform_refs:
+                    raise KeyError(
+                        "neither job nor task transforms may be specified unless you also specify a loader"
```

Comment:
Maybe change this to `f"Default transforms ({t}) cannot be specified unless..."`. That way we don't hardcode the names of the default transforms.